static/test/javascripts/spec/common/commercial/third-party-tags/outbrain.spec.js

```diff
@@ -66,151 +74,141 @@ define([
+            it('should load when user is logged in but there are no comments on the page', function (done) {
                 identity.isUserLoggedIn = function () {
                     return true;
                 };
 
                 config.page.commentable = false;
-                spyOn(sut, 'load');
 
-                sut.init();
+                sut.init().then(function () {
+                    expect(sut.load).toHaveBeenCalled();
+                    done();
+                });
                 mediator.emit('modules:commercial:dfp:rendered', eventStub);
-                expect(sut.load).toHaveBeenCalled();
+                mediator.emit('modules:commercial:dfp:rendered', eventStubLo);
             });
 
-            /*
-                Loading Outbrain is dependent on succefull return of high relevance component
-                from DFP. AdBlock is blocking DFP calls so we are not getting any response and thus
-                not loading Outbrain. As Outbrain is being partially loaded behind the adblock we can
-                make the call instantly when we detect adBlock in use.
-             */
-            it('should load instantly when ad block is in use', function () {
-                spyOn(sut, 'load');
-
+            it('should load instantly when ad block is in use', function (done) {
                 detect.adblockInUse = function () { return true; };
 
-                sut.init();
-                expect(sut.load).toHaveBeenCalled();
+                sut.init().then(function () {
+                    expect(sut.load).toHaveBeenCalled();
+                    done();
+                });
             });
 
-            it('should load instantly when international edition', function () {
-                spyOn(sut, 'load');
+            it('should load in the low-priority merch component', function (done) {
+                eventStub.isEmpty = false;
+                eventStubLo.isEmpty = true;
+                config.switches.outbrainReplacesMerch = true;
+
+                var oldEmit = mediator.emit;
+                mediator.emit = function (eventName, data) {
+                    return new Promise(function (resolve) {
+                        oldEmit.call(mediator, eventName, data);
+                        resolve();
+                    });
+                };
 
-                config.page.edition = 'INT';
+                sut.init().then(function () {
+                    expect(sut.load).toHaveBeenCalledWith('merchandising');
+                    done();
+                });
+                mediator.emit('modules:commercial:dfp:rendered', eventStub).then(function () {
+                    mediator.emit('modules:commercial:dfp:rendered', eventStubLo);
+                });
 
-                sut.init();
-                expect(sut.load).toHaveBeenCalled();
+                mediator.emit = oldEmit;
             });
         });
 
```

Comment:
We should include test for the case when high and low priority are both on the page. To make sure that in that case we definitely not load outbrain.